src/payment/strategies/clearpay/clearpay-payment-strategy.ts

```diff
@@ -0,0 +1,106 @@
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { paymentMethods: { getPaymentMethodOrThrow } } = this._store.getState();
+        const paymentMethod = getPaymentMethodOrThrow(options.methodId, options.gatewayId);
+
+        this._clearpaySdk = await this._clearpayScriptLoader.load(paymentMethod);
+
+        return this._store.getState();
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._clearpaySdk = undefined;
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const paymentId = payload.payment && payload.payment.gatewayId;
+
+        if (!paymentId) {
+            throw new PaymentArgumentInvalidError(['payment.gatewayId']);
+        }
+
+        const { isStoreCreditApplied: useStoreCredit } = this._store.getState().checkout.getCheckoutOrThrow();
+        let state = this._store.getState();
+
+        if (useStoreCredit !== undefined) {
+            state = await this._store.dispatch(this._storeCreditActionCreator.applyStoreCredit(useStoreCredit));
+        }
+
+         // Validate nothing has changed before redirecting to Clearpay checkout page
+        await this._checkoutValidator.validate(state.checkout.getCheckout(), options);
+
+        state = await this._store.dispatch(
+            this._paymentMethodActionCreator.loadPaymentMethod(paymentId, options)
+        );
+
+        this._redirectToClearpay(state.paymentMethods.getPaymentMethod(paymentId));
+
+        // Clearpay will handle the rest of the flow so return a promise that doesn't really resolve
+        return new Promise<never>(() => {});
+    }
+
+    async finalize(options: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const state = this._store.getState();
+        const payment = state.payment.getPaymentId();
+        const config = state.config.getContextConfig();
+
+        if (!payment) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckout);
+        }
+
+        if (!config || !config.payment.token) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+        }
+
+        const paymentPayload = {
+            methodId: payment.providerId,
+            paymentData: { nonce: config.payment.token },
+        };
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder({}, options));
+
+        return this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
+    }
+
+    private _redirectToClearpay(paymentMethod?: PaymentMethod): void {
+        if (!this._clearpaySdk || !paymentMethod || !paymentMethod.clientToken) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        this._clearpaySdk.initialize({countryCode: this._mapCountryToISO2()});
```

Comment:
This is not a mapping function so the name is misleading. I wonder if we should use the country code comes from the billing address? And if it's not UK show an error message?

Reply:
I think the problem here is that a shopper could have selected a transactional currency that is no GBP, and that would mean that suddenly if they choose Clearpay, everything will be in GBP.

Even if Clearpay doesn't allow it, we should cover that scenario and handle it properly, rather than hardcoding the country ourselves. This will help developers using the SDK in case they make a mistake.